### PR TITLE
Documentation for Jetstream TLS configuration

### LINKED ISSUE #2697: Add documentation for pubsub/jetstream tls configuration
> **Note**: maybe there needs to be a new issue template (or reword an existing template) for simple updates?

**What content needs to be created or modified?**

Document new fields for TLS Client Authentication.

**Describe the solution you'd like**

N/A

**Where should the new material be placed?**

Existing

**The associated pull request from dapr/dapr, dapr/components-contrib, or other Dapr code repos**

https://github.com/dapr/components-contrib/pull/1924

**Additional context**

N/A

### PR DESCRIPTION
**Pull request by NickLarsenNZ**
Thank you for helping make the Dapr documentation better!

**Please follow this checklist before submitting:**
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://docs.dapr.io/contributing/contributing-overview/#developer-certificate-of-origin-signing-your-work))
- [x] [Read the contribution guide](https://docs.dapr.io/contributing/contributing-docs/)
- [ ] Commands include options for Linux, MacOS, and Windows within codetabs
- [ ] New file and folder names are globally unique
- [ ] Page references use shortcodes instead of markdown or URL links
- [ ] Images use HTML style and have alternative text
- [ ] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

## Description

- Added TLS configuration fields for client authentication.
- Formatted the markdown table for improved readability and easier editing.
- Made example values consistent (string quotes inside the code span).

## Issue reference

closes https://github.com/dapr/docs/issues/2697

### CODE REVIEW

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-jetstream.md [hunk 1]
```diff
@@ -46,18 +46,20 @@ spec:
...
-| startSequence  |        N | [Start Sequence] | `1` |
-| startTime      |        N | [Start Time] in Unix format | `1630349391` |
-| deliverAll     |        N | Set deliver all as [Replay Policy] | `true` |
-| flowControl    |        N | [Flow Control] | `true` |
+| Field           | Required | Details                                    | Example                          |
+| --------------- | :------: | ------------------------------------------ | -------------------------------- |
+| natsURL         |    Y     | NATS server address URL                    | `"nats://localhost:4222"`        |
+| jwt             |    N     | NATS decentralized authentication JWT      | `"eyJhbGciOiJ...6yJV_adQssw5c"`  |
+| seedKey         |    N     | NATS decentralized authentication seed key | `"SUACS34K232O...5Z3POU7BNIL4Y"` |
+| tls_client_cert |    N     | NATS TLS Client Authentication Certificate | `"/path/to/tls.crt"`             |
```
Comment (ItalyPaleAle) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-jetstream.md:58]: Can you please add these fields to the sample YAML above too?

**Comment by yaron2** [comment]
Ping @NickLarsenNZ

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-jetstream.md (hunk 1 above)
Comment (NickLarsenNZ) [reply daprdocs/content/en/reference/components-reference/supported-pubsub/setup-jetstream.md:58]: I think they are mutually exclusive with jwt auth. Should I make a separate sample config?

**Comment by NickLarsenNZ** [comment]
Hi @ItalyPaleAle, are you able to open this PR and reply to my comment above?
FYI @yaron2

**Comment by msfussell** [comment]
Reopening wrongly closed PR!

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-jetstream.md (hunk 1 above)
Comment (ItalyPaleAle) [reply daprdocs/content/en/reference/components-reference/supported-pubsub/setup-jetstream.md:58]: You can list them as optional (with a `#` at the beginning)

**Comment by ItalyPaleAle** [comment]
@NickLarsenNZ Sorry I missed this, I responded now (but ultimately it's @msfussell 's choice)

**Comment by msfussell** [comment]
@NickLarsenNZ - Apologies for missing this, due to it being closed. The suggestion of creating a separate a sample config is a good one. You can see another example of this here https://docs.dapr.io/reference/components-reference/supported-pubsub/setup-azure-eventhubs/. I suggest having the fields in the table for the description with a separate sample configure under a heading of using TLS or similar. Makes sense?

**Comment by msfussell** [comment]
Ping @NickLarsenNZ

**Comment by NickLarsenNZ** [comment]
Thanks @msfussell , I'll take a look this weekend.

**Comment by NickLarsenNZ** [comment]
> The suggestion of creating a separate a sample config is a good one. You can see another example of this here [docs.dapr.io/reference/components-reference/supported-pubsub/setup-azure-eventhubs](https://docs.dapr.io/reference/components-reference/supported-pubsub/setup-azure-eventhubs/). I suggest having the fields in the table for the description with a separate sample configure under a heading of using TLS or similar. Makes sense?

@msfussell would you mind if I made the sample configurations tabbed?
As I've been working with the Jetstream PubSub component, I've come to understand the options a lot better and I think a few additional config samples with explanations might help others. IMO, the tabs look nicer than a longer page and help people jump straight to what they're looking for while hiding less relevant stuff.

The configuration tabs I had in mind were:
- `Basic`
- `Decentralized JWT Authentication`
- `TLS Client Authentication`
- `Durable Consumers`
- `Queue Group`
- `Consumer Offset`

**Comment by hhunter-ms** [comment]
> > The suggestion of creating a separate a sample config is a good one. You can see another example of this here [docs.dapr.io/reference/components-reference/supported-pubsub/setup-azure-eventhubs](https://docs.dapr.io/reference/components-reference/supported-pubsub/setup-azure-eventhubs/). I suggest having the fields in the table for the description with a separate sample configure under a heading of using TLS or similar. Makes sense?
> 
> @msfussell would you mind if I made the sample configurations tabbed? As I've been working with the Jetstream PubSub component, I've come to understand the options a lot better and I think a few additional config samples with explanations might help others. IMO, the tabs look nicer than a longer page and help people jump straight to what they're looking for while hiding less relevant stuff.
> 
> The configuration tabs I had in mind were:
> 
> * `Basic`
> * `Decentralized JWT Authentication`
> * `TLS Client Authentication`
> * `Durable Consumers`
> * `Queue Group`
> * `Consumer Offset`

@NickLarsenNZ I'll create a new docs issue for this, as a post-release docs enhancement.

**Review by msfussell — APPROVED** [review]
LGTM

**Pull request merged**